group-generators/helpers/data-operators/map.ts

```diff
@@ -1,15 +1,44 @@
-import { BigNumberish } from "ethers";
+import { BigNumber, BigNumberish } from "ethers";
 import { FetchedData } from "topics/group";
 
+export type Thresholds = {
+  min: number;
+  newValue: number;
+}[]
+
 export const Map = (
   groupData: FetchedData,
-  value: BigNumberish
+  value?: BigNumberish,
```

Comment:
i think this is a good feature to have 👍 
however, now the map interface is a bit unclear
i suggest we keep the old map as it was and create a new map that allows more complicated mapping schemes
something like this
```
enum MapCriteria {
  LT = 'LT',
  GT = 'GT',
  EQ = 'EQ',
}

type MapOptions = Array<{
  criteria: MapCriteria;
  newValue: number;
}>;

const options: MapOptions = [
  { criteria: MapCriteria.LT, newValue: 10 },
  { criteria: MapCriteria.GT, newValue: 20 },
  { criteria: MapCriteria.EQ, newValue: 30 },
];
```